src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,434 @@
+
+                await this.ExecuteCommandAsync($"./arm-performance-libraries_22.1_Ubuntu-20.04.sh", $"-a", armPackageLibrariesPath, true, telemetryContext, cancellationToken);
+            }
+
+            HPLinpackState state = await this.stateManager.GetStateAsync<HPLinpackState>($"{nameof(HPLinpackState)}", cancellationToken)
+                ?? new HPLinpackState();
+
+            if (!state.HPLInitialized)
+            {
+                await this.ExecuteCommandAsync("wget", $"http://www.netlib.org/benchmark/hpl/hpl-{this.Version}.tar.gz -O {this.PackageName}.tar.gz", this.PlatformSpecifics.PackagesDirectory, false, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("tar", $"-zxvf {this.PackageName}.tar.gz", this.PlatformSpecifics.PackagesDirectory, false, telemetryContext, cancellationToken);
+                state.HPLInitialized = true;
+
+            }
+
+            this.HPLDirectory = this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, $"hpl-{this.Version}");
+            await this.stateManager.SaveStateAsync<HPLinpackState>($"{nameof(HPLinpackState)}", state, cancellationToken);
+
+            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, this.makeFileName))
+                .ConfigureAwait(false);
+
+            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, "setup", this.makeFileName))
+                .ConfigureAwait(false);
+
+            await this.ExecuteCommandAsync("bash", "-c \"source make_generic\"", this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup"), true, telemetryContext, cancellationToken);
+
+            await this.ConfigureMakeFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            await this.ExecuteCommandAsync("ln", $"-s {this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup", this.makeFileName)} {this.makeFileName}", this.HPLDirectory, false, telemetryContext, cancellationToken);
+
+        }
+
+        /// <summary>
+        /// Executes the HPL workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            DateTime startTime = DateTime.UtcNow;
+            await this.ExecuteCommandAsync("make", $"arch=Linux_GCC", this.HPLDirectory, false, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.SetParameters();
+            await this.ConfigureDatFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            string results;
+            if (this.CpuArchitecture == Architecture.X64)
+            {
+                results = await this.ExecuteCommandAsync("runuser", $"-u {this.Username} -- mpirun --use-hwthread-cpus -np {this.coreCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), true, telemetryContext, cancellationToken)
```

Comment:
Are the spaces between -- and mpirun intentional?

Reply:
runuser [options] -u <user> [[--] <command>]
runuser [options] [-] [<user> [<argument>...]]

Run <command> with the effective user ID and group ID of <user>. 
The options -c, -f, -l, and -s are mutually exclusive with -u.
where - c is command , -f,-l,-s are few other parameters for runuser. It's the syntax of runuser and we can't use -u and -c parameters at once. These are mutually exclusive.